src/prefect/tasks.py

```diff
@@ -850,16 +901,69 @@ def submit(
                 return from_sync.wait_for_call_in_loop_thread(
                     create_autonomous_task_run_call
                 )
+        if PREFECT_EXPERIMENTAL_ENABLE_NEW_ENGINE and flow_run_context and self.isasync:
+            return self._submit_async(
+                parameters=parameters,
+                flow_run_context=flow_run_context,
+                wait_for=wait_for,
+                return_state=return_state,
+            )
 
-        return enter_task_run_engine(
-            self,
+        else:
+            return enter_task_run_engine(
+                self,
+                parameters=parameters,
+                wait_for=wait_for,
+                return_type=return_type,
+                task_runner=None,  # Use the flow's task runner
+                mapped=False,
+            )
+
+    async def _submit_async(
+        self,
+        parameters: Dict[str, Any],
+        flow_run_context: FlowRunContext,
+        wait_for: Optional[Iterable[PrefectFuture]],
+        return_state: bool,
+    ):
+        from prefect.new_task_engine import run_task
+
+        task_runner = flow_run_context.task_runner
+
+        task_run = await self.create_run(
+            flow_run_context=flow_run_context,
             parameters=parameters,
             wait_for=wait_for,
-            return_type=return_type,
-            task_runner=None,  # Use the flow's task runner
-            mapped=False,
         )
 
+        future = PrefectFuture(
+            name=task_run.name,
+            key=uuid4(),
+            task_runner=task_runner,
+            asynchronous=(self.isasync and flow_run_context.flow.isasync),
+        )
+        future.task_run = task_run
+        flow_run_context.task_run_futures.append(future)
+        await task_runner.submit(
+            key=future.key,
+            call=partial(
+                run_task,
+                task=self,
+                task_run=task_run,
+                parameters=parameters,
+                wait_for=wait_for,
+                return_type="state",
+            ),
+        )
+        # TODO: I don't like this. Can we move responsibility for creating the future
+        # and setting this anyio.Event to the task runner?
+        future._submitted.set()
+
+        if return_state:
+            return await future._wait()
```

Comment:
does this block until completion?

Reply:
Ah, yes it will. Let me try with `.get_state()` instead

Reply:
It turns out that to match the current functionality, we need to call `.wait` because `return_state` returns the result of the task wrapped in a state.

Reply:
Yea gotcha - that makes sense